api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -763,9 +765,26 @@ public User setUserActivationKey(User user) {
 		LoginCredential credentials = dao.getLoginCredential(user);
 		credentials.setActivationKey(activationKey);	
 		dao.setUserActivationKey(credentials);	
-			
-		//Send Email with unhashed  activation key and Date:
 		
+		String link = Context.getAdministrationService().getGlobalProperty(OpenmrsConstants.GP_HOST_URL) + token;
+		String msg = "Hi " + user.getUsername()
+		        + ", \n It looks like you like to change your password, click on the following link to do so.\n" + link
+		        + " please disregard this email if you didn't innitiate the password reset process  The openmrs Team";
+		Message mail = new Message();
```

Comment:
You should not be constructing anything, please use the message service, all this should be like 2 calls to the service once you have the contents of the email.